scalafix-core/src/main/scala/scalafix/util/Patch.scala

```diff
@@ -30,7 +32,16 @@ abstract class ImportPatch(val importer: Importer) extends TreePatch {
 }
 
 object TreePatch {
-  case class Rename(from: Name, to: Name) extends TreePatch
+  trait RenamePatch
+  case class Rename(from: Name, to: Name) extends TreePatch with RenamePatch
+  case class RenameSymbol(from: Symbol, to: Name, normalize: Boolean = false)
```

Comment:
@ShaneDelmore FYI, this tree patch works similarly to "Rename" in IDEs. I'm thinking of renaming `Rename` to `RenameReference` or `RenameThisInstance`, what do you think?

I plan to document the patches better soon. First, I want to focus on getting rid of the ScalafixMirror and improve build integrations.